### ISSUE TITLE
Clang and OpenMP

### ISSUE THREAD

**Issue by Psirus** [nutofem/nuto#105]
Currently, we're unable to use OpenMP in combination with Clang. [Starting with 3.8](https://openmp.llvm.org/), Clang supports OpenMP. Previous builds of Clang+OpenMP did not use OpenMP (!), as it is not found and then ignored by CMake (see e.g. [this build](https://travis-ci.org/nutofem/nuto/jobs/271983476) "-- Could NOT find OpenMP"). This wasn't a problem before, except for the useless build duplication, but with [an update to the Travis images](https://blog.travis-ci.com/2017-08-29-trusty-image-updates), Clang 3.9 is shipped. Now, CMake does find OpenMP, but the runtime library `libomp.so` is not linked to our executables, and the tests fail.

### Write own FindOpenMP

On Travis, Clang is installed from the [tarball releases of LLVM](http://releases.llvm.org/), so `libomp.so` is probably installed somewhere as well, we just need to find it and link to it. Possible, but no fun.

### Use CMake 3.9

With the current CMake version, [FindOpenMP](https://cmake.org/cmake/help/v3.9/module/FindOpenMP.html) has been bored up to return the library paths as well (previous versions didn't). Also, it get's the nice `IMPORTED` targets syntax. And while installing CMake 3.9 on Travis would be okay, I think either a) forcing everyone to upgrade to 3.9 now or b) maintaining some sort of `if CMake ≥ 3.9: foo; else: bar` in our CMake code is not desirable.

### Wait

For now, I'd say we leave it at that, and once CMake 3.9 becomes universally available (whatever that means), we do the above.

**Comment by phuschke** [comment]
Thank you for the information. 

I vote for **Use CMake 3.9 a)** and have peace.

**Comment by Psirus** [comment]
If we were to go that way, installing CMake 3.9 under distros that don't have in in there repos yet goes like this:

    $ wget https://cmake.org/files/v3.9/cmake-3.9.2-Linux-x86_64.sh
    $ sudo sh cmake-3.9.2-Linux-x86_64.sh -- --skip-license --prefix=/usr

**Comment by TTitscher** [comment]
Cmake3.9 is too far away from debian stable and ubuntu stable. An update will at least take approx two years or so, right?
And since we want to stay compatible with those versions, I vote against the required manual install of version 3.9.

As for the original problem: Locally, NuTo with clang and openmp runs smoothly without manually providing libomp.so. So is this (just) a problem with travis? Could we set the path for travis manually?

**Comment by Psirus** [comment]
"Locally", the libomp.so of Clang ends up in a standard path when installing from the repos. On Travis, Clang has been installed using the tarballs from llvm.org, but I'm not sure where to. If you want, find that path and then I guess you would have to supply it to LD_LIBRARY_PATH or something.

**Comment by rtobar** [comment]
Indeed, setting `LD_LIBRARY_PATH=/usr/local/clang-3.9.0/lib` in your environment does work (just fixed my build). Executables are actually linked (otherwise they would fail to compile), but the library is not found at runtime because it's located in this odd location.
